lib/steps.js

```diff
@@ -52,20 +52,36 @@ let click_with = {
 };
 
 BeforeAll(async () => {
+  scope.timeout = 120 * 1000;
+  scope.showif_timeout = 500;
   scope.report = {};
-
 });
 
 Before(async (scenario) => {
-  scope.timeout = 120 * 1000;
-  scope.showif_timeout = 500;
-  scope.scenario_name = scenario.pickle.name.replace(/[^A-Za-z0-9]/gi, '');
+  // Names
+  let scenario_name = scenario.pickle.name.replace(/[^A-Za-z0-9]/gi, '');
+  let tags = ''
+  for ( let tag of scenario.pickle.tags ) {
+    tags += tag.name.replace(/[^A-Za-z0-9]+/g, '_').replace(/(\P{L})+/gu, '_');
+  }
+  // With tag names automatically added to langauges, this should work to make
+  // each language unique.
+  scope.safe_id = `_${ scenario_name }${ tags }`
+
+  // Should we add a counter? It would have to be unique to each scenario name.
+  if ( scope.report[ scope.safe_id ] ) {
+    scope.safe_id += `_1`;
```

Comment:
Hmm... I'm not sure this works in all cases. What if you have keys `abc` and `abc_1` in `scope.report` already, and the current `scope.safe_id = 'abc'`? I think this would end up creating a duplicate key in that case. 

Not sure if this is worth worrying about right now, since this could be a rare edge case that we can afford to ignore for now.

Reply:
You're totally right. I _knew_ there was something wrong with that. I didn't want to bother adding a count, though.

Bleh, if it's not going to work properly should we just remove it and leave a note saying that we'll take care of it later? It seems useless to make it iterate and just keep adding `_1` anyway. We could just add an underscore each time... Something like this:
```
...
while ( scope.report[ new_id ] !== undefined  ) {
  new_id += '_';
}
scope.safe_id = new_id;
```

It doesn't seem very clear, though.

Reply:
I think the above code snippet will work, and we could add a comment explaining what the code snippet does. We don't need to add the new code snippet in right now, but I do think we should remove what currently exists, or make a ticket explaining it needs to be removed.

Reply:
Eh, I did it anyway. It's not ideal, but it is what it is.